pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -738,16 +738,17 @@ func (nc *Controller) monitorNodeHealth(ctx context.Context) error {
 
 			_, needsRetry := nc.nodesToRetry.Load(node.Name)
 			switch {
-			case currentReadyCondition.Status != v1.ConditionTrue && observedReadyCondition.Status == v1.ConditionTrue:
```

Comment:
After this change, the event will not be reported if status is changed. Is this Correct?